### ISSUE TITLE
Unable to Output </CONCLUSION> Correctly

### ISSUE THREAD

**Issue by xlxcomputer** [shawnricecake/Heima#2]
I am very excited to see your latest work, and I find it incredibly interesting. However, when I tried to reproduce the results of the Encoder (MLLM) in the demo, I encountered some issues. I am unable to obtain the correct </CONCLUSION> output. My output is as follows:
<img width="393" alt="Image" src="https://github.com/user-attachments/assets/f17ade7d-8758-4435-b255-33dc42529dc9" />
This does not correctly answer the question, and I consistently get the same result after multiple runs. This has left me quite puzzled. Since I only used the encoder for testing, I made some modifications to the configuration file (5-llama3_2_vision-11b-demo-decode-pure_llm_decoder_lora-split_3_stages.yaml) and the code (5-demo-decode-pure_llm_decoder_lora-split_3_stages.py). However, after double-checking, I don’t believe these changes are likely to cause the issue, as Heima implements decoding for both the Encoder and Decoder processes, and using only the Encoder part should still work.

**I suspect the issue might be related to the following:**

The tokenizer.model (encoder) path provided in the configuration file is an absolute path, which I assume points to the llava-cot tokenizer. However, I couldn’t find the corresponding tokenizer.model file in the Hugging Face repository for llava-cot. As a result, I am currently using the tokenizer from meta-llama/Llama-3.2-11B-Vision-Instruct (since we know llava-cot is fine-tuned from Llama-3.2-11B-Vision-Instruct).

**In summary, I would like to seek your assistance in two areas:**

- Since most NLP users rely on the transformers library for training and inference rather than Torchtune, it’s unclear how to obtain the llava-cot tokenizer. The Hugging Face repository for llava-cot does not include the tokenizer.model file. Could you please upload the tokenizer.model file directly to the repository to facilitate easier reproduction for the community?

- Do you have any insights into why the MLLM is only outputting </CONCLUSION>Yes</CONCLUSION>? Do you have any solution?

I would greatly appreciate your help with this matter. Thank you in advance!

<img width="573" alt="Image" src="https://github.com/user-attachments/assets/d8ac899e-1900-42ee-88da-628c7f8dae71" />

![Image](https://github.com/user-attachments/assets/b01912e4-3e04-49cc-a3ff-9c0798d67f39)

**Comment by shawnricecake** [comment]
try different seed

**Comment by xlxcomputer** [comment]
Thank you so much for your help! I have solved my problem!

At the same time, I have discovered some **interesting phenomena** in Heima that I would like to discuss with you. I found that when I set the seed to certain specific values, **such as seed=6 or seed=32, the model always responds with just "YES"**.

However, when I set **seed=14**, the response appears as shown in the image below. The model seems to generate a lot of redundant information and reaches the **maximum token limit** (max_token=128).
<img width="826" alt="Image" src="https://github.com/user-attachments/assets/6628a6a7-f5f5-4c5d-9a7b-816899abdb47" />
When I set **seed=42**, the response appears as shown in the image below. The model correctly completed my answer.

<img width="586" alt="Image" src="https://github.com/user-attachments/assets/125e4bf7-fbc7-4fc6-8e85-55b363f4f633" />

**Does this suggest that Heima's responses are highly sensitive to the seed?** (I'm not sure whether this is caused by LLava-CoT itself.)

**Comment by shawnricecake** [comment]
I am surprised by the results with **seed=14**.

and I have no idea with the sensitivity of seed to the model performance.

by the way, **seed=42** is a magic seed, you can check out in `The Hitchhiker’s Guide to the Galaxy`. :laughing:

**Comment by xlxcomputer** [comment]
hahaha😀
Thanks for your help!
Bset wishes🤞
